app/data-store/reducers/calendar-reducer.js

```diff
@@ -16,14 +16,16 @@ export default (state = {}, action) => {
 
         case 'NAVIGATE':
             if (state.router.route == 'index') {
-                // page is now ready, initialize the calendar...
-                setTimeout(() => {
-                    renderCalendar();
-                },100);
-
+                setTimeout(renderCalendar,1,state.sources);
+                setTimeout(renderEventsOnNavigate,1,state.sources);
             }
             return state;
 
+        case 'TOGGLEROOM':
+
+            Object.assign({}, state, state.sources[action.value.room - 1].visible = !action.value.visible );
```

Comment:
I get it now I need to alter the state object then return the new altered state object. Sorry Just figuring this out.